pkg/build/controller/buildconfig/buildconfig_controller.go

```diff
@@ -174,9 +175,52 @@ func (c *BuildConfigController) buildConfigAdded(obj interface{}) {
 // buildconfig is updated or there is a relist of buildconfigs
 func (c *BuildConfigController) buildConfigUpdated(old, cur interface{}) {
 	bc := cur.(*buildv1.BuildConfig)
+	oldBuildConfig := old.(*buildv1.BuildConfig)
+	// TODO: Remove the event after the image trigger behavior is removed.
+	if c.imageChangeTriggerCleared(oldBuildConfig, bc) {
+		c.recorder.Event(bc,
+			corev1.EventTypeWarning,
+			"ImageChangeTriggerCleared",
+			"spec.triggers[*].imagechange.lastTriggeredImageID was cleared, which will trigger a build. This behavior is deprecated and will be removed in a future OpenShift release.")
+	}
 	c.enqueueBuildConfig(bc)
 }
 
+func (c *BuildConfigController) imageChangeTriggerCleared(old, cur *buildv1.BuildConfig) bool {
+	if old == nil || cur == nil {
+		return false
+	}
+	imageLastTriggers := map[corev1.ObjectReference]string{}
+	for _, oldTrigger := range old.Spec.Triggers {
+		if oldTrigger.ImageChange == nil {
+			continue
+		}
+		from := c.getImageChangeTriggerInputReference(old, oldTrigger)
+		imageLastTriggers[*from] = oldTrigger.ImageChange.LastTriggeredImageID
+	}
+	for _, currentTrigger := range cur.Spec.Triggers {
+		if currentTrigger.ImageChange == nil {
+			continue
+		}
+		from := c.getImageChangeTriggerInputReference(cur, currentTrigger)
+		prev, found := imageLastTriggers[*from]
+		if found && len(prev) > 0 && len(currentTrigger.ImageChange.LastTriggeredImageID) == 0 {
```

Comment:
do we care if the devops user set it to some non-zero "dummy" value vs. clearing it out

i.e. alert that they should not be setting it in any fashion

Reply:
I think we care less about this scenario, but is something we can check. If the current is an invalid image ref, we have a pretty good idea that the image trigger controller didn't do this.

Reply:
Actually, I don't think we can reliably do this other than with the empty string. "foo" is a perfectly valid pull spec - otherwise would would need to assume that the image trigger always adds the full SHA reference to this field.

Reply:
the validity of the image ref was not a much a thought for me as the fact that they really should not be setting it at all IMO

that said, upon further review, it could get kludgy at best figuring out if a user set it vs. the controller set it

that, and don't feel strongly about it really anyway ... so let's move on .... an edge case at most, if a concern at all, so we can wait for it to crop up in a customer case before worrying about it at most